src/plugins/dashboard/public/application/utils/direct_query_sync/direct_query_sync.ts

```diff
@@ -0,0 +1,183 @@
+    // The "null state" for a fresh page load, which components conditionally use on load.
+    fresh: { ord: 100, terminal: true },
+  })
+);
+
+export const MAX_ORD = 100;
+
+export function intervalAsMinutes(interval: number): string {
+  const minutes = Math.floor(interval / 60000);
+  return minutes === 1 ? '1 minute' : minutes + ' minutes';
+}
+
+export async function resolveConcreteIndex(
+  indexTitle: string,
+  http: HttpStart,
+  mdsId?: string
+): Promise<string | null> {
+  if (!indexTitle.includes('*')) return indexTitle;
+
+  try {
+    const query: any = mdsId ? { data_source: mdsId } : {};
+    const resolved = await http.get(
+      `/internal/index-pattern-management/resolve_index/${encodeURIComponent(indexTitle)}`,
+      { query }
+    );
+    const matchedIndices = resolved?.indices || [];
+    return matchedIndices.length > 0 ? matchedIndices[0].name : null;
+  } catch (err) {
+    return null;
+  }
+}
+
+export function extractIndexParts(fullIndexName: string): IndexExtractionResult {
+  const trimmed = fullIndexName.replace(/^flint_/, '');
+  const parts = trimmed.split('_');
+  return {
+    datasource: parts[0] || 'unknown',
+    database: parts[1] || 'unknown',
+    index: parts.slice(2).join('_') || 'unknown',
+  };
+}
+
+export function generateRefreshQuery(info: IndexExtractionResult): string {
+  return `REFRESH MATERIALIZED VIEW \`${info.datasource}\`.\`${info.database}\`.\`${info.index}\``;
+}
+
+export async function extractIndexInfoFromDashboard(
+  panels: { [key: string]: any },
+  savedObjectsClient: SavedObjectsClientContract,
+  http: HttpStart
+): Promise<{
+  parts: IndexExtractionResult;
+  mapping: { lastRefreshTime: number };
+  mdsId?: string;
+} | null> {
+  const indexPatternIds: string[] = [];
+  const mdsIds: Array<string | undefined> = [];
+
+  for (const panelId of Object.keys(panels)) {
+    try {
+      const panel = panels[panelId];
+      const savedObjectId = panel.explicitInput?.savedObjectId;
+      if (!savedObjectId) continue;
+
+      const type = panel.type;
+      const savedObject = await savedObjectsClient.get(type, savedObjectId);
+
+      const references = savedObject.references || [];
+
+      if (references.length === 0) {
+        continue;
+      }
+
+      // Check if there is any non-index-pattern reference
+      if (references.some((ref: any) => ref.type !== 'index-pattern')) {
+        return null;
+      }
+
+      const indexPatternRef = references.find((ref: any) => ref.type === 'index-pattern');
+      if (!indexPatternRef) {
+        return null;
+      }
+
+      const indexPattern = await savedObjectsClient.get('index-pattern', indexPatternRef.id);
+      const mdsId =
+        indexPattern.references?.find((ref: any) => ref.type === 'data-source')?.id || undefined;
+
+      indexPatternIds.push(indexPatternRef.id);
+      mdsIds.push(mdsId);
+    } catch (err) {
+      // Ignoring error: saved object might be missing or invalid
```

Comment:
can we not silently swallow all types of error? If it's known type of error, use if conditions to only swallow those specific error types

Reply:
fixed